api/src/main/java/org/openmrs/api/impl/DiagnosisServiceImpl.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+import org.openmrs.api.APIException;
+import org.openmrs.api.DiagnosisService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.DiagnosisDAO;
+import org.openmrs.parameter.EncounterSearchCriteria;
+import org.openmrs.parameter.EncounterSearchCriteriaBuilder;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+
+@Transactional
+public class DiagnosisServiceImpl extends BaseOpenmrsService implements DiagnosisService {
+	
+	private DiagnosisDAO diagnosisDAO;
+
+	/**
+	 * Saves a diagnosis
+	 *
+	 * @param diagnosis - the diagnosis to be saved
+	 * @return the diagnosis
+	 */
+	@Override
+	public Diagnosis save(Diagnosis diagnosis) {
+		diagnosisDAO.saveDiagnosis(diagnosis);
+		return diagnosis;
+	}
+
+	/**
+	 * Voids a diagnosis
+	 *
+	 * @param diagnosis  - the diagnosis to be voided
+	 * @param voidReason - the reason for voiding the diagnosis
+	 * @return the diagnosis that was voided
+	 */
+	@Override
+	public Diagnosis voidDiagnosis(Diagnosis diagnosis, String voidReason) {
+		return Context.getDiagnosisService().save(diagnosis);
+	}
+
+	/**
+	 * Gets a diagnosis based on the uuid
+	 *
+	 * @param uuid - uuid of the diagnosis to be returned
+	 * @return diagnosis matching the given uuid
+	 */
+	@Override
+	public Diagnosis getDiagnosisByUuid(String uuid) {
+		return diagnosisDAO.getDiagnosisByUuid(uuid);
+	}
+
+	/**
+	 * Gets diagnoses since date, sorted in reverse chronological order
+	 *
+	 * @param patient  the patient whose diagnosis we are to get
+	 * @param fromDate the date used to filter diagnosis which happened from this date and later
+	 * @return the list of diagnoses for the given patient and starting from the given date
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Diagnosis> getDiagnoses(Patient patient, Date fromDate) {
+		return diagnosisDAO.getActiveDiagnoses(patient);
```

Comment:
Did you intentionally ignore the fromDate?

Reply:
No I didn't. I just added this